### PR TITLE
H2DB store wire component checks on user input

### PR DESCRIPTION
**Pull request by marcellorinaldo**
Added some checks to the user-definable properties `maximum.table.size` and `cleanup.records.keep`.

**Related Issue:** N/A.

**Description of the solution adopted:** Prior to this PR, if the user set `cleanup.records.keep > maximum.table.size` the cleanup to the DB tables won't happen. In particular, the table truncation was executed by selecting the oldest `maximum.table.size - cleanup.records.keep` entries, which could assume negative values causing an error. Several checks are now implemented and the minimum value for `maximum.table.size` property is set to 1.

Unit tests have been refactored according to Gherkin style and updated with new corner-case checks.

**Screenshots:** N/A.

**Any side note on the changes made:** N/A.

### CODE REVIEW

**Comment by MMaiero** [comment]
@mattdibi Can you please have a review of this PR? Thanks

File: kura/org.eclipse.kura.wire.h2db.component.provider/src/main/java/org/eclipse/kura/internal/wire/h2db/store/H2DbWireRecordStore.java [hunk 1]
```diff
@@ -89,6 +89,9 @@
 
     private static final String[] TABLE_TYPE = new String[] { "TABLE" };
 
+    private static final String NULL_TABLE_NAME_ERROR_MSG = "Table name cannot be null";
+    private static final String NULL_WIRE_RECORD_ERROR_MSG = "";
```
Comment (mattdibi) [inline kura/org.eclipse.kura.wire.h2db.component.provider/src/main/java/org/eclipse/kura/internal/wire/h2db/store/H2DbWireRecordStore.java]: Is this supposed to be empty?

Comment (marcellorinaldo) [reply kura/org.eclipse.kura.wire.h2db.component.provider/src/main/java/org/eclipse/kura/internal/wire/h2db/store/H2DbWireRecordStore.java]: Forgot to readd the error message :)

**Review by salvatore-coppola — APPROVED** [review]

**Review by mattdibi — APPROVED** [review]
LGTM
